### PR TITLE
Adding info on entities selected to catalog table title

### PR DESCRIPTION
**Pull request by SnowBlitzer**
## Hey, I just made a Pull Request!

In prep for user research, we realized this page can cause confusion on what is currently selected. This makes the title on the table more descriptive for the user.
Screenshot 2023-05-17 at 10 52 01 AM

Screenshot 2023-05-17 at 10 51 42 AM
Screenshot 2023-05-17 at 10 51 54 AM

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by efenner-cambia** [comment]
This looks great!

**Review by Rugvip — CHANGES_REQUESTED** [review]
Ah sorry, missed actually posting the review 🤦

Thank you! Looks good, just a small nit

File: plugins/catalog/src/components/CatalogTable/CatalogTable.tsx [hunk 1]
```diff
@@ -241,7 +246,7 @@ export const CatalogTable = (props: CatalogTableProps) => {
         pageSizeOptions: [20, 50, 100],
         ...tableOptions,
       }}
-      title={`${titlePreamble} (${entities.length})`}
+      title={`${titleDisplay}${currentKind ? 's' : ''} (${entities.length})`}
```
[the remark this answers is not in the document]
Comment (SnowBlitzer) [reply plugins/catalog/src/components/CatalogTable/CatalogTable.tsx:249]: Ohhh, thank you for this. I'm having an issue pushing the commit but I will get that up

**Review by Rugvip — APPROVED** [review]
Nice! 👍 

Let's :shipit:

**Pull request merged**